### ISSUE TITLE
[Extensibility Request] Expose "Telemetry User ID"

### ISSUE THREAD

**Issue by jwikman** [microsoft/ALAppExtensions#25202]
### Describe the request

Please expose the value of the `Telemetry User ID` field from the table `User Property` in some function.
It is now hidden from SaaS by an OnPrem scope on the table (which is reasonable on that table).

### Additional context

The `User Property` table has a OnPrem Scope, which is reasonable considered what is stored in that table, except for one field - `Telemetry User ID`.

The only place where `Telemetry User ID` is used in any app, is on the User Card, so user can clear or modify its value. This value is than added by the platform to the default BC telemetry that is emitted in the `user_id` field of the telemetry. This is really valuable in some troubleshooting when we need to identify a user that has some strange behaviors seen in telemetry.

When we emit our own telemetry, via the `Session.LogMessage()` function, the `user_id` above is not included so we had to implement our own clone of the `Telemetry User ID` field.

Soon it will also be easier to find a user from this `user_id` value, when the `Telemetry User ID` field will be shown in a list (see https://github.com/microsoft/BCApps/pull/254, which hopefully makes it into BC24). This will probably increase the use of the `Telemetry User ID` field.

Another approach of solving this is that the platform adds `user_id` also to telemetry created from the `Session.LogMessage()` function.

**Comment by IhorHandziuk** [comment]
Hi @jwikman , it is intentional that the `user_id` field is not included in IsV telemetry store (defined in app.json) - it is due to compliance reasons.
This field should be included in the environment telemetry that your apps emit with TelemetryScope::All. Is that not the case?

**Comment by jwikman** [comment]
Hi @IhorHandziuk 

What are the compliance reasons?

As an ISV, we cannot do anything with the `user_id` without the help of the customer or partner that has access to the environment. But when we got an issue from a customer or partner, it would be very helpful to be able to filter out the particular user from our telemetry.

> This field should be included in the environment telemetry that your apps emit with TelemetryScope::All though. Is that not the case?

Yes, that is probably true - but we do not send everything to the environment telemetry. And we, as an ISV, cannot access that telemetry anyway.

**Comment by KennieNP** [comment]
As for compliance, we have decided that the user id is only emitted to telemetry for the environment as this is close to the partner/admin that deals directly with the customer.

**Comment by jwikman** [comment]
That is unfortunate, and I cannot see the reason for that decision since the ISV cannot do anything with the user id without the help of a partner.
It only complicates things when ISV tries to help partner out.

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.

Build ID: <None>.

**Comment by jwikman** [comment]
@JesperSchulz I'm trying to figure out what you agreed on here 😅

Can we expect the `user_id` to be added to ISV telemetry?

**Comment by JesperSchulz** [comment]
Let me consult @KennieNP to get an answer to your question 🙂

**Comment by KennieNP** [comment]
user_id will likely not be added to ISV telemetry. At least not until we have a capability for the administrator who install an app to opt-in/out of sending user_id to ISV telemetry.

**Comment by jwikman** [comment]
Okay, that's not what I wanted to hear. 😉

I take that as that this won't happen in a quite a long time then.
